packages/form-js-editor/src/features/properties-panel/entries/HeadersSourceSelectEntry.js

```diff
@@ -0,0 +1,123 @@
+import { isSelectEntryEdited } from '@bpmn-io/properties-panel';
+
+import { AutoFocusSelectEntry } from '../components';
+
+import { get, isString, isArray } from 'min-dash';
+
+
+const OPTIONS = {
+  static: {
+    label: 'List of items',
+    value: 'static'
+  },
+  expression: {
+    label: 'Expression',
+    value: 'expression'
+  }
+};
+
+const SELECT_OPTIONS = Object.values(OPTIONS);
+
+const COLUMNS_PATH = [ 'columns' ];
+
+const COLUMNS_EXPRESSION_PATH = [ 'columnsExpression' ];
+
+export function HeadersSourceSelectEntry(props) {
+  const {
+    editField,
+    field,
+    id
+  } = props;
+
+  return [
+    {
+      id: id + '-select',
+      component: HeadersSourceSelect,
+      isEdited: isSelectEntryEdited,
+      editField,
+      field
+    }
+  ];
+}
+
+function HeadersSourceSelect(props) {
+
+  const {
+    editField,
+    field,
+    id
+  } = props;
+
+  /**
+   * @returns {string|void}
+   */
+  const getValue = () => {
+    const columns = get(field, COLUMNS_PATH);
+    const columnsExpression = get(field, COLUMNS_EXPRESSION_PATH);
+
+    if (isString(columnsExpression)) {
+      return OPTIONS.expression.value;
+    }
+
+    if (isArray(columns)) {
+      return OPTIONS.static.value;
+    }
+  };
+
+  /**
+   * @param {string|void} value
+   */
+  const setValue = (value) => {
+    switch (value) {
+    case OPTIONS.static.value:
+      editField(field, {
+        columnsExpression: undefined,
```

Comment:
For `values` we have a [modeling behavior](https://github.com/bpmn-io/form-js/blob/develop/packages/form-js-editor/src/features/modeling/behavior/ValuesSourceBehavior.js#L29) that removes the other properties. The advantage of that approach is that it also does the clean ups when using the modeling API directly, and not only in the properties panel entry. It's also pluggable as an extension can simply remove that behavior (for whatever reason). 

Wdyt, should we do the same for `columnsExpression` and `columns` or only once it becomes more complex with more possibilities do define the source?

Reply:
Yeah, I was waiting for you to raise this topic. I'll take a look into this tomorrow

Reply:
@pinussilvestrus Should be done now

Reply:
Thanks for the follow up, looks good to me 👍